Zend/zend_compile.c

```diff
@@ -4801,6 +4801,9 @@ static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args)
 			ZVAL_STRINGL(&const_node.u.constant, offset, p - offset);
 			opline = zend_compile_rope_add(result, rope_elements++, &const_node);
 		}
+		if (elem_node.op_type == IS_CONST) {
+			convert_to_string(&elem_node.u.constant);
```

Comment:
I'm not sure we can do this at compile time when the constant coercion emits an warning, e.g. for arrays. You might have to use `_try_convert_to_string`, and not use the optimization at all otherwise.

Reply:
Could steal from `zend_compile_binary_op()` for `ZEND_CONCAT`.